core/formats/wand_writer.hpp

```diff
@@ -54,40 +56,48 @@ class WandWriterImpl final : public WandWriter {
     score_t score;
     producer_.GetScore(&score);
 
-    Update(score_levels_.front(), score,
-           [&]() { return producer_.GetValue(); });
+    Update(score_levels_.front(), score, [&] { return producer_.GetValue(); });
   }
 
   void Write(size_t level, memory_index_output& out) final {
-    IRS_ASSERT(level < score_levels_.size());
+    IRS_ASSERT(level + 1 < score_levels_.size());
     auto& score = score_levels_[level];
-    if (level < score_levels_.size() - 1) {
-      // Accumulate score on less granular level
-      Update(score_levels_[level + 1], score.score,
-             [&]() { return score.value; });
-
-      IRS_ASSERT(std::is_sorted(score_levels_.begin(),
-                                score_levels_.begin() + level + 2,
-                                [](const auto& lhs, const auto& rhs) noexcept {
-                                  return lhs.score < rhs.score;
-                                }));
-    }
+    // Accumulate score on less granular level
+    Update(score_levels_[level + 1], score.score, [&] { return score.value; });
+
+    IRS_ASSERT(
+      std::is_sorted(score_levels_.begin(), score_levels_.begin() + level + 2));
     ValueProducer::Write(score.value, out);
     score.score = 0.f;
   }
 
+  void WriteRoot(index_output& out) final {
+    ValueProducer::Write(score_levels_.back().value, out);
+  }
+
   byte_type Size(size_t level) const noexcept final {
-    IRS_ASSERT(level < score_levels_.size());
+    IRS_ASSERT(level + 1 < score_levels_.size());
     const auto& entry = score_levels_[level];
     return ValueProducer::Size(entry.value);
   }
 
+  byte_type SizeRoot() noexcept final {
+    auto max = std::max_element(score_levels_.begin(), score_levels_.end());
+    IRS_ASSERT(max != score_levels_.end());
```

Comment:
std::max_element always returns valid iterator unless range.
```suggestion
    IRS_ASSERT(!score_levels_.empty());
```

Reply:
Fixed